app/models/retention_schedule.rb

```diff
@@ -34,7 +39,7 @@ class RetentionSchedule < ApplicationRecord
       transitions from: [:retain], to: :not_set
     end
 
-    event :anonymise! do
```

Comment:
I removed the bang because AASM treats [bang events](https://github.com/aasm/aasm#bang-events) differently, in particular they trigger the auto-save of the object, and also updates timestamp columns automagically.